app/Http/Controllers/Api/Places/SpecialFeaturesController.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Hedonist\Http\Controllers\Api\Places;
+
+use Hedonist\Http\Controllers\Api\ApiController;
+use Hedonist\Http\Requests\Place\SpecialFeature\CreateSpecialFeatureHttpRequest;
+
+use Hedonist\Actions\Place\SpecialFeature\{
+    CreateSpecialFeatureAction,
+    DeleteSpecialFeatureAction,
+    ReadSpecialFeatureAction,
+    CollectionSpecialFeatureAction
+};
+use Hedonist\Actions\Place\SpecialFeature\{
+    CreateSpecialFeatureRequest,
+    DeleteSpecialFeatureRequest,
+    ReadSpecialFeatureRequest,
+    CollectionSpecialFeatureRequest
+};
+use Hedonist\Actions\Place\SpecialFeature\{
+    CreateSpecialFeatureResponse,
+    DeleteSpecialFeatureResponse,
+    ReadSpecialFeatureResponse,
+    CollectionSpecialFeatureResponse
+};
+
+class SpecialFeaturesController extends ApiController
+{
+    /** @var CreateSpecialFeatureAction **/
+    private $createSpecialFeatureAction;
+    /** @var DeleteSpecialFeatureAction **/
+    private $deleteSpecialFeatureAction;
+    /** @var ReadSpecialFeatureAction **/
+    private $readSpecialFeatureAction;
+    /** @var CollectionSpecialFeatureAction **/
+    private $collectionSpecialFeatureAction;
+
+    /**
+     * SpecialFeaturesController constructor.
+     * @param CreateSpecialFeatureAction $createSpecialFeatureAction
+     * @param DeleteSpecialFeatureAction $deleteSpecialFeatureAction
+     * @param ReadSpecialFeatureAction $readSpecialFeatureAction
+     * @param CollectionSpecialFeatureAction $collectionSpecialFeatureAction
+     */
+    public function __construct(
+        CreateSpecialFeatureAction $createSpecialFeatureAction,
+        DeleteSpecialFeatureAction $deleteSpecialFeatureAction,
+        ReadSpecialFeatureAction $readSpecialFeatureAction,
+        CollectionSpecialFeatureAction $collectionSpecialFeatureAction
+    )
+    {
+        $this->createSpecialFeatureAction = $createSpecialFeatureAction;
+        $this->deleteSpecialFeatureAction = $deleteSpecialFeatureAction;
+        $this->readSpecialFeatureAction = $readSpecialFeatureAction;
+        $this->collectionSpecialFeatureAction = $collectionSpecialFeatureAction;
+    }
+
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function index()
+    {
+        try {
+            /** @var CollectionSpecialFeatureResponse $response */
+            $response = $this->collectionSpecialFeatureAction->execute(
+                new CollectionSpecialFeatureRequest()
+            );
+        } catch (\Exception $ex) {
+            // base api controller
+            return $this->errorResponse($ex->getMessage(), 400);
+        }
+
+        $responseArray = [];
```

Comment:
Move this logic to `CollectionSpecialFeaturePresenter`, or better you can implement [JsonSerializable](http://php.net/manual/ru/class.jsonserializable.php) on response